odl/operator/default_ops.py

```diff
@@ -791,8 +765,9 @@ def __init__(self, constant, domain=None, range=None):
         if range is None:
             range = constant.space
 
-        super().__init__(domain, range)
-        self.__constant = self.range.element(constant)
+        self.__constant = range.element(constant)
+        linear = self.constant.norm() == 0
```

Comment:
Read this through, nowhere is the operator implied to work on scalars, so this wont work. We have `ConstantFunctional` for that.